### DISCUSSION TITLE
Is it possible to let Quickadd send a hotkey or a hotkey string to Obsidian?

### DISCUSSION THREAD

**Discussion by xmbai9507** [chhoumann/quickadd#659] [Q&A]
For example, if I need to move the cursor left by one character, i.e. pressing the left arrow key once, how can I integrate this into the macro? 
Or in another case, if there is a command that pops up an input box and receives users' desired playback speed of the video being watched, when added to a macro, how can I automatically type in a string like 'x2.25' when this command is executed?

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

QuickAdd doesn't send synthetic keystrokes, and it won't - Electron largely ignores untrusted key events, so it would be brittle even if we tried. But both of your actual goals are reachable:

**1) Moving the cursor left by one character.** Use a user script step in the macro and drive the editor API directly instead of simulating ArrowLeft:

```js
module.exports = async ({ app }) => {
  const editor = app.workspace.activeEditor?.editor;
  if (!editor) return;
  const c = editor.getCursor();
  editor.setCursor({ line: c.line, ch: Math.max(0, c.ch - 1) });
};
```

**2) Auto-typing "x2.25" into another plugin's input box.** QuickAdd can run any Obsidian command as a macro step (Macro builder -> Obsidian Command), or from a script via `app.commands.executeCommandById(...)`. What it can't do is type into another plugin's modal once it opens. If that plugin exposes an API or a command that takes the value, you can call it from a user script via `app.plugins.plugins["<plugin-id>"]`; if it only offers an interactive input box, the clean fix is to ask that plugin's author for a parameterized command.

More on user script steps in macros: https://quickadd.obsidian.guide/docs/Choices/MacroChoice
